transfermethodui/src/androidTest/java/com/hyperwallet/android/ui/transfermethod/SelectTransferMethodTest.java

```diff
@@ -446,4 +492,45 @@ public void testSelectTransferMethod_verifyDefaultsToUSWhenUserProfileDoesNotHav
         onView(withId(R.id.select_transfer_method_country_value)).check(matches(withText("United States")));
         onView(withId(R.id.select_transfer_method_currency_value)).check(matches(withText("USD")));
     }
+
+    @Test
+    public void testSelectTransferMethod_verifyClickEventOnSameCountrySelection() {
+        mMockWebServer.mockResponse().withHttpResponseCode(HTTP_OK).withBody(sResourceManager
+                .getResourceContent("user_response.json")).mock();
+        mMockWebServer.mockResponse().withHttpResponseCode(HTTP_OK).withBody(sResourceManager
+                .getResourceContent("successful_tmc_keys_response.json")).mock();
+
+        mActivityTestRule.launchActivity(null);
+
+        onView(withId(R.id.select_transfer_method_country_value)).perform(click());
+        onView(allOf(withId(R.id.country_name), withText("United States"))).perform(click());
+
+        verify(mHyperwalletInsight, atMost(1)).trackClick(any(Context.class), pageNameCaptor.capture(),
+                pageGroupCaptor.capture(), linkCaptor.capture(),
+                mapClickCaptor.capture());
+
+        assertEquals("transfer-method:add:select-transfer-method", pageNameCaptor.getValue());
+        assertEquals("transfer-method", pageGroupCaptor.getValue());
+        assertEquals("select-country", linkCaptor.getValue());
+        assertEquals(1, mapClickCaptor.getAllValues().get(0).size());
+        assertEquals("US", mapClickCaptor.getAllValues().get(0).get(
+                InsightEventTag.InsightEventTagEventParams.TRANSFER_METHOD_COUNTRY));
+    }
+
+    @Test
+    public void testSelectTransferMethod_verifySendEventWhenUserClicksOnCountrySelectionButNavigationBack() {
+        mMockWebServer.mockResponse().withHttpResponseCode(HTTP_OK).withBody(sResourceManager
+                .getResourceContent("user_response.json")).mock();
+        mMockWebServer.mockResponse().withHttpResponseCode(HTTP_OK).withBody(sResourceManager
+                .getResourceContent("successful_tmc_keys_large_response.json")).mock();
+
+        mActivityTestRule.launchActivity(null);
+
+        onView(withId(R.id.select_transfer_method_country_container)).perform(click());
+        UiDevice mDevice = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation());
+        mDevice.pressBack();
+
+        verify(mHyperwalletInsight, never()).trackClick(any(Context.class), anyString(),
```

Comment:
@skoong As I have created same method on PR #114 for another ticket, please use this line alone in the same method while merging. Thanks in advance.